code/modules/mob/living/carbon/human/species.dm

```diff
@@ -1610,87 +1610,233 @@ GLOBAL_LIST_EMPTY(roundstart_races)
-		// clear any hot moods and apply cold mood
-		SEND_SIGNAL(H, COMSIG_CLEAR_MOOD_EVENT, "hot")
-		SEND_SIGNAL(H, COMSIG_ADD_MOOD_EVENT, "cold", /datum/mood_event/cold)
-		// Apply cold slow down
-		H.add_or_update_variable_movespeed_modifier(/datum/movespeed_modifier/cold, multiplicative_slowdown = ((bodytemp_cold_damage_limit - H.bodytemperature) / COLD_SLOWDOWN_FACTOR))
-		// Display alerts based on the amount of cold damage being taken
-		// Apply more damage based on how cold you are
-		switch(H.bodytemperature)
-			if(200 to bodytemp_cold_damage_limit)
-				H.throw_alert("temp", /obj/screen/alert/cold, 1)
-				H.apply_damage(COLD_DAMAGE_LEVEL_1 * coldmod * H.physiology.cold_mod, BURN)
+	// Apply some burn damage to the body
+	if(humi.coretemperature < bodytemp_cold_damage_limit && !HAS_TRAIT(humi, TRAIT_RESISTCOLD))
+		switch(humi.coretemperature)
+			if(201 to bodytemp_cold_damage_limit)
+				humi.apply_damage(COLD_DAMAGE_LEVEL_1 * coldmod * humi.physiology.cold_mod, BURN)
 			if(120 to 200)
-				H.throw_alert("temp", /obj/screen/alert/cold, 2)
-				H.apply_damage(COLD_DAMAGE_LEVEL_2 * coldmod * H.physiology.cold_mod, BURN)
+				humi.apply_damage(COLD_DAMAGE_LEVEL_2 * coldmod * humi.physiology.cold_mod, BURN)
 			else
-				H.throw_alert("temp", /obj/screen/alert/cold, 3)
-				H.apply_damage(COLD_DAMAGE_LEVEL_3 * coldmod * H.physiology.cold_mod, BURN)
+				humi.apply_damage(COLD_DAMAGE_LEVEL_3 * coldmod * humi.physiology.cold_mod, BURN)
 
-	// We are not to hot or cold, remove status and moods
-	else
-		H.clear_alert("temp")
-		H.remove_movespeed_modifier(/datum/movespeed_modifier/cold)
-		SEND_SIGNAL(H, COMSIG_CLEAR_MOOD_EVENT, "cold")
-		SEND_SIGNAL(H, COMSIG_CLEAR_MOOD_EVENT, "hot")
+/**
+ * Used to apply burn wounds on random limbs
+ *
+ * This is called from body_temperature_damage when exposure to extream heat adds up and causes a wound.
+ * The wounds will increase in severity as the temperature increases.
+ * vars:
+ * * humi (required) The mob we will targeting
+ */
+/datum/species/proc/apply_burn_wounds(mob/living/carbon/human/humi)
+	if(HAS_TRAIT(humi, TRAIT_RESISTHEAT))
+		return
+
+	// Skin temp is very high we need to apply some burns
+	if(humi.bodytemperature > bodytemp_heat_damage_limit)
+		var/datum/wound/burn_type
+		var/burn_damage = HEAT_DAMAGE_LEVEL_1
+
+		switch(humi.bodytemperature)
+			if(0 to 400)
+				burn_type = /datum/wound/burn
+			if(401 to 600)
+				burn_type = /datum/wound/burn/moderate
+				burn_damage = HEAT_DAMAGE_LEVEL_2
+			if(601 to 2200)
+				burn_type = /datum/wound/burn/severe
+				burn_damage = HEAT_DAMAGE_LEVEL_2
+			else
+				burn_type = /datum/wound/burn/critical
+				burn_damage = HEAT_DAMAGE_LEVEL_3
+
+		var/obj/item/bodypart/bodypart = pick(humi.bodyparts)
+		var/datum/wound/burn/existing_burn = locate(/datum/wound/burn) in bodypart.wounds
+
+		//Apply the burn wounds, replace existing burns we do not want stacks
```

Comment:
This behavior is already implemented in /obj/item/bodypart/proc/force_wound_upwards(), see https://github.com/tgstation/tgstation/blob/af65c90125502f8ccae91622bb8653ba9c76dbe6/code/modules/surgery/bodyparts/_bodyparts.dm#L434

Reply:
This made things much better thanks.